test/unit/math/prim/prob/multi_student_t_cholesky_test.cpp

```diff
@@ -0,0 +1,461 @@
+  double nu = 4.0;
+
+  EXPECT_THROW(multi_student_t_cholesky_lpdf(y, nu, mu, L),
+               std::invalid_argument);
+  EXPECT_THROW(multi_student_t_cholesky_rng(nu, mu, L, rng),
+               std::invalid_argument);
+}
+
+TEST(ProbDistributionsMultiStudentTCholesky, ErrorSize3) {
+  using Eigen::Dynamic;
+  using Eigen::Matrix;
+  using stan::math::multi_student_t_cholesky_lpdf;
+  using stan::math::multi_student_t_cholesky_rng;
+  using std::vector;
+  boost::random::mt19937 rng;
+  Matrix<double, Dynamic, 1> y(3, 1);
+  y << 2.0, -2.0, 11.0;
+  Matrix<double, Dynamic, 1> mu(3, 1);
+  mu << 1.0, -1.0, 3.0;
+  Matrix<double, Dynamic, Dynamic> Sigma(4, 4);
+  Sigma << 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0,
+      0.0, 1.0;
+  Matrix<double, Dynamic, Dynamic> L = Sigma.llt().matrixL();
+  double nu = 4.0;
+
+  EXPECT_THROW(multi_student_t_cholesky_lpdf(y, nu, mu, L),
+               std::invalid_argument);
+  EXPECT_THROW(multi_student_t_cholesky_rng(nu, mu, L, rng),
+               std::invalid_argument);
+}
+
+TEST(ProbDistributionsMultiStudentTCholesky, ProptoAllDoublesZero) {
+  using Eigen::Dynamic;
+  using Eigen::Matrix;
+  using stan::math::multi_student_t_cholesky_lpdf;
+  using stan::math::multi_student_t_rng;
+  using std::vector;
+  Matrix<double, Dynamic, 1> y(3, 1);
+  y << 2.0, -2.0, 11.0;
+  Matrix<double, Dynamic, 1> mu(3, 1);
+  mu << 1.0, -1.0, 3.0;
+  Matrix<double, Dynamic, Dynamic> Sigma(3, 3);
+  Sigma << 9.0, -3.0, 0.0, -3.0, 4.0, 0.0, 0.0, 0.0, 5.0;
+  Matrix<double, Dynamic, Dynamic> L = Sigma.llt().matrixL();
+  double nu = 4.0;
+
+  EXPECT_FLOAT_EQ(0.0, multi_student_t_cholesky_lpdf<true>(y, nu, mu, L));
+}
+
+TEST(ProbDistributionsMultiStudentTCholesky,
+     marginalOneChiSquareGoodnessFitTest) {
+  using Eigen::Dynamic;
+  using Eigen::Matrix;
+  using stan::math::multi_student_t_cholesky_rng;
+  using std::vector;
+  boost::random::mt19937 rng;
+  Matrix<double, Dynamic, 1> mu(3, 1);
+  mu << 2.0, 3.0, 11.0;
+
+  Matrix<double, Dynamic, Dynamic> s(3, 3);
+  s << 10.0, 3.0, 11.0, 3.0, 9.0, 1.2, 11.0, 1.2, 16.0;
+  Matrix<double, Dynamic, Dynamic> L = s.llt().matrixL();
+  int N = 10000;
+  int K = stan::math::round(2 * std::pow(N, 0.4));
+  boost::math::students_t_distribution<> dist(3.0);
+  boost::math::chi_squared mydist(K - 1);
+
+  double loc[K - 1];
+  for (int i = 1; i < K; i++)
+    loc[i - 1] = quantile(dist, i * std::pow(K, -1.0));
+
+  int count = 0;
+  int bin[K];
+  double expect[K];
+  for (int i = 0; i < K; i++) {
+    bin[i] = 0;
+    expect[i] = N / K;
+  }
+
+  Eigen::VectorXd a(mu.rows());
+  while (count < N) {
+    a = multi_student_t_cholesky_rng(3.0, mu, L, rng);
+    a(0) = (a(0) - mu(0, 0)) / std::sqrt(s(0, 0));
+    int i = 0;
+    while (i < K - 1 && a(0) > loc[i])
+      ++i;
+    ++bin[i];
+    count++;
+  }
+
+  double chi = 0;
+  for (int j = 0; j < K; j++)
+    chi += ((bin[j] - expect[j]) * (bin[j] - expect[j]) / expect[j]);
```

Comment:
[optional]

If `bin` and `expect` are declared as Eigen vectors (and set to `Constant` to initialize), this can be done with vector operations, but it's messy, so I wouldn't recommend.  On the other hand, it makes it easy to at least compute `bin - expect` once.  But no big deal inside these tests.  

The empirical statistic is conventionally called `X`, not `chi` (don't ask me---I didn't invent stats' whacky notation).

Reply:
done